src/middlewared/middlewared/service.py

```diff
@@ -871,6 +872,32 @@ def ping(self):
         """
         return 'pong'
 
+    def _ping_host(self, host, count, timeout):
+        if osc.IS_LINUX:
+            process = run(['ping', '-c', f'{count}', '-w', f'{timeout}', host])
+        else:
+            process = run(['ping', '-c', f'{count}', '-t', f'{timeout}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    @accepts(
+        Str('hostname'),
+        Int('timeout', default=10),  # seconds
+    )
+    def ping_remote(self, hostname, timeout):
+        """
+        Utility method call a ping in the other node.
+        The other node will just return "pong".
+        """
+        ping_host = self._ping_host(hostname, 1, timeout)
```

Comment:
If we're pinging a remote system that we've never talked to before, then it's almost guaranteed to drop that packet because we don't have an ARP address for that IP or the switch has to update it's arp table.. I'd change the count to at least 2 probably 3.

Reply:
@yocalebo you have a good point! If there is no MAC address in ARP table, the first probably fails. Changed.